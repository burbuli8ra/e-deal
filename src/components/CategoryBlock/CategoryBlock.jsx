import React from 'react';
import Styled from './CategoryBlock.style';

const CategoryBlock = ({
  id,
  imageUrl,
  title
}) => {
  return (
    <Styled.CategoryBlock data-testid="categoryItem-block" to={`/categories/${id}`}>
      <Styled.Image alt={title} src={imageUrl} />
      {title}
    </Styled.CategoryBlock>
  );
};

export default CategoryBlock;