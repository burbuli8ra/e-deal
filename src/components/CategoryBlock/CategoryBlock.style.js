import styled from '@emotion/styled';
import { Link } from 'react-router-dom';
import { colors } from 'theme';

const CategoryBlock = styled.article`
  margin: 10px;
`;
CategoryBlock.displayName = 'CategoryBlock';

const CategoryBlockLink = styled(Link)`
  align-items: center;
  background-color: ${colors.white};
  border-radius: 4px;
  box-shadow: 0 0 9px 0 rgba(140, 140, 158, 0.25);
  color: ${colors.darkPastelBlue};
  display: inline-flex;
  flex-direction: column;
  font-weight: 700;
  max-width: 300px;
  overflow: hidden;
  padding: 20px;
  text-decoration: none;
`;
CategoryBlockLink.displayName = 'CategoryBlockLink';

const Image = styled.img`
  height: auto;
  width: 100%;
  margin-bottom: 10px;
`;
Image.displayName = 'Image';

export default {
  CategoryBlock,
  CategoryBlockLink,
  Image
};