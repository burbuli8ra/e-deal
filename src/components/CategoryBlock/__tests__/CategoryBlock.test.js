import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Mocks from '__mocks__';
import CategoryBlock from '../CategoryBlock';

const setup = props => (
  <Router>
    <CategoryBlock {...props} />
  </Router>
);

describe('CategoryBlock component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = Mocks.CategoryItem;
    wrapper = render(setup(props));
  });

  test('should match snapshot', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  });
});