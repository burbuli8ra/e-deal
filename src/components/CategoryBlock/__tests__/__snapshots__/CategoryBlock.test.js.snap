// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryBlock component should match snapshot 1`] = `
<DocumentFragment>
  <article
    class="css-1lu5s7y"
    data-testid="categoryItem-block"
    to="/categories/6989"
  >
    <a
      class="css-1ttiqzk"
      href="/categories/6989"
    >
      <img
        alt="Τεχνολογία"
        class="css-jd31iv"
        src="https://abpcdn.pstatic.gr/P/bpimg128/6989_SX400Y400/1577836800/technology.webp"
      />
      Τεχνολογία
    </a>
  </article>
</DocumentFragment>
`;
